server/coordinator/procedure/ddl/createpartitiontable/create_partition_table.go

```diff
@@ -219,35 +219,62 @@ func createDataTablesCallback(event *fsm.Event) {
 	}
 
 	shardVersions := req.p.relatedVersionInfo.ShardWithVersion
+	shardTableMetaDatas := make(map[storage.ShardID][]metadata.CreateTableMetadataRequest, 0)
 	for i, subTableShard := range params.SubTablesShards {
-		result, err := params.ClusterMetadata.CreateTableMetadata(req.ctx, metadata.CreateTableMetadataRequest{
+		tableMetaData := metadata.CreateTableMetadataRequest{
 			SchemaName:    params.SourceReq.GetSchemaName(),
 			TableName:     params.SourceReq.GetPartitionTableInfo().SubTableNames[i],
 			PartitionInfo: storage.PartitionInfo{},
-		})
+		}
+		shardTableMetaDatas[subTableShard.ShardInfo.ID] = append(shardTableMetaDatas[subTableShard.ShardInfo.ID], tableMetaData)
+	}
+	shardVersionsCh := make(map[storage.ShardID]chan uint64, len(shardTableMetaDatas))
+	for shardID, tableMetaDatas := range shardTableMetaDatas {
+		shardVersion := shardVersions[shardID]
+		shardVersionsCh[shardID] = make(chan uint64)
+		go createDataTables(event, shardID, tableMetaDatas, shardVersion, shardVersionsCh[shardID])
```

Comment:
What will happen if invoke `createDataTables` failed? It seems that we could not get error and stop creating table. Maybe we need use `errorgroup` to get `createDataTables` result.